tools/hsmtool.c

```diff
@@ -450,147 +396,70 @@ static int guess_to_remote(const char *address, struct node_id *node_id,
+	/* Get mnemonic from user using consistent interface */
+	mnemonic = read_stdin_mnemonic(tmpctx, &err);
+	if (!mnemonic)
+		errx(EXITCODE_ERROR_HSM_FILE, "Could not read mnemonic: %s", hsm_secret_error_str(err));
 
-	printf("Introduce your BIP39 word list separated by space (at least 12 words):\n");
+	/* Get optional passphrase */
+	printf("Warning: remember that different passphrases yield different "
+	       "bitcoin wallets.\n");
+	printf("If left empty, no password is used (echo is disabled).\n");
+	printf("Enter your passphrase: \n");
 	fflush(stdout);
-	size_t characters = getline(&line, &line_size, stdin);
-	if (characters < 0)
-		errx(ERROR_USAGE, "Could not read line from stdin.");
-	line[characters-1] = '\0';
-	return line;
-}
-
-static char *read_mnemonic(void) {
-	/* Get words for the mnemonic language */
-	struct words *words;
-	get_words(&words);
-
-	/* Get mnemonic */
-	char *mnemonic;
-	mnemonic = get_mnemonic();
-
-	if (bip39_mnemonic_validate(words, mnemonic) != 0) {
-		errx(ERROR_USAGE, "Invalid mnemonic: \"%s\"", mnemonic);
-	}
-	return mnemonic;
-}
-
-static int generate_hsm(const char *hsm_secret_path,
-			const char *lang_id,
-			char *mnemonic,
-			char *passphrase)
-{
-	const char *err;
-	int exit_code = 0;
-
-	if (lang_id == NULL) {
-		mnemonic = read_mnemonic();
-		printf("Warning: remember that different passphrases yield different "
-		       "bitcoin wallets.\n");
-		printf("If left empty, no password is used (echo is disabled).\n");
-		printf("Enter your passphrase: \n");
-		fflush(stdout);
-		passphrase = read_stdin_pass_with_exit_code(&err, &exit_code);
-		if (!passphrase)
-			errx(exit_code, "%s", err);
-		if (strlen(passphrase) == 0) {
-			free(passphrase);
-			passphrase = NULL;
-		}
-	} else {
-		struct words *words;
-
-		bip39_get_wordlist(lang_id, &words);
-
-		if (bip39_mnemonic_validate(words, mnemonic) != 0)
-			errx(ERROR_USAGE, "Invalid mnemonic: \"%s\"", mnemonic);
+	passphrase = read_stdin_pass(tmpctx, &err);
+	if (!passphrase)
+		errx(EXITCODE_ERROR_HSM_FILE, "Could not read passphrase: %s", hsm_secret_error_str(err));
+	if (strlen(passphrase) == 0) {
+		passphrase = NULL;
 	}
 
-	u8 bip32_seed[BIP39_SEED_LEN_512];
-	size_t bip32_seed_len;
-
-	if (bip39_mnemonic_to_seed(mnemonic, passphrase, bip32_seed, sizeof(bip32_seed), &bip32_seed_len) != WALLY_OK)
-		errx(ERROR_LIBWALLY, "Unable to derive BIP32 seed from BIP39 mnemonic");
-
+	/* Write to file using your new mnemonic format */
 	int fd = open(hsm_secret_path, O_CREAT|O_EXCL|O_WRONLY, 0400);
 	if (fd < 0) {
 		errx(ERROR_USAGE, "Unable to create hsm_secret file");
 	}
-	/* Write only the first 32 bytes, length of the (plaintext) seed in the
-	 * hsm_secret. */
-	if (!write_all(fd, bip32_seed, 32))
-		errx(ERROR_USAGE, "Error writing secret to hsm_secret file");
+	
+	if (passphrase) {
+		/* Write passphrase hash (32 bytes) + mnemonic for protected format */
+		struct sha256 sha;
+		sha256(&sha, passphrase, strlen(passphrase));
```

Comment:
This should be hash of the final seed.  This is a bit trickier: we need hsm_secret to expose that "calculate seed from mnemonic and passphrase" function (or opencode it with libwally?)